conf/config.go

```diff
@@ -0,0 +1,193 @@
+   files, for the BCDA app. This package will go through three different stages.
+
+   1. Local env looks primarily at conf package for variables, but will also look
+   in the environment for any variables it is not tracking. PROD/TEST/DEV will
+   only look in the environment. (WE ARE HERE NOW)
+
+   2. Local env will only look at conf package. PROD/TEST/DEV will look at both.
+
+   3. All env will look at the conf package.
+
+   Assumptions:
+   1. The configuration file is a env file (can be changed if needed).
+   2. The configuration variables, once it has been ingested by the conf package,
+   will stay immutable during the uptime of the application (exception is test).
+*/
+
+import (
+	"os"
+	"testing"
+
+	"github.com/spf13/viper"
+)
+
+// Private global variable:
+
+// An instance of the viper struct containing the config information. Only made
+// accessible through public functions: GetEnv, SetEnv, etc.
+var envVars viper.Viper
+
+// Implementing a state machine that tracks the status of the config ingestion.
+// This state machine should go away when in stage 3.
+const (
+	configgood    uint8 = 0
+	configbad     uint8 = 1
+	noconfigfound uint8 = 2
+)
+
+// if config file found and loaded, doesn't changed
+var state uint8 = configgood
+
+/*
+   This is the private helper function that sets up viper. This function is
+   called by the init() function only once during initialization of the package.
+*/
+func setup(dir string) *viper.Viper {
+
+	// Viper setup
+	var v = viper.New()
+	v.SetConfigName("local")
+	v.SetConfigType("env")
+	v.AddConfigPath(dir)
+	// Viper is lazy, do the read and parse of the config file
+	var err = v.ReadInConfig()
+
+	// If viper cannot read the configuration file...
+	if err != nil {
+		state = configbad
+	}
+
+	return v
+
+}
+
+/*
+   init:
+   First thing to run when this package is loaded by the binary.
+   Even if multiple packages import conf, this will be called and ran ONLY once.
+*/
+func init() {
+
+	// Possible config file locations: local and PROD/DEV/TEST respectfully.
+	var locationSlice = [2]string{
+		"/go/src/github.com/CMSgov/bcda-app/shared_files/decrypted",
+		"/go/src/github.com/CMSgov/DoesNotExistYet", // This is a placeholder for now
+	}
+
+	if success, loc := findEnv(locationSlice[:]); success {
+		// A config file found, set up viper using that location
+		envVars = *setup(loc)
+	} else {
+		// Checked both locations, no config file found
+		state = noconfigfound
+	}
+
+}
+
+/*
+   findEnv is a helper function that will determine what environment the application
+   is running in: local or PROD/TEST/DEV. Each environment should have a distinct
+   path where the configuration file is located. First it check the local path,
+   then the PROD/DEV/TEST. If both not found, defaults to just using environment vars.
+*/
+func findEnv(location []string) (bool, string) {
+
+	// Check if the configuration file exists
+	if _, err := os.Stat(location[0] + "/local.env"); err == nil {
```

Comment:
Would it be clearer for us to do: 
```
for _, loc := range location {
if _, err := os.Stat(location[0] + "/local.env"); err == nil {
return true, loc
}
}

return false, ""
```

Reply:
That way you do not have your exit condition explicitly check for the end of the list. It happens for free with the range.